src/hotspot/share/opto/superword.cpp

```diff
@@ -521,10 +521,14 @@ bool SuperWord::SLP_extract() {
 
   combine_pairs_to_longer_packs();
 
-  split_packs_longer_than_max_vector_size();
+  construct_my_pack_map();
+
+  split_packs_at_use_def_boundaries();  // a first time: create natural boundaries
+  split_packs_only_implemented_with_smaller_size();
+  split_packs_to_break_mutual_dependence();
+  split_packs_at_use_def_boundaries();  // again: propagate split of other packs
 
   // Now we only remove packs:
-  construct_my_pack_map();
```

Comment:
Note: as with the `filter_packs` methods, I will for now just leave all `split_packs` methods here. I can clean this up in a future RFE. Things around here may still change, be added and removed. For now it is most convenient to keep it all here in a simple list.

Also: I now moved `construct_my_pack_map` before the filtering. This allows the filters to have access to `my_pack`. I need that for the `use_def_boundaries` check.